packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -158,7 +159,8 @@ func GetBlockProofStatus(
 		}, nil
 	}
 
-	if proverAddress == transition.Prover {
+	if proverAddress == transition.Prover ||
+		(proverSetAddress != ZeroAddress && proverAddress == proverSetAddress) {
```

Comment:
Would this explain the block's proof has already been submitted？

Reply:
Yep if a prover is behind a proverSet, the proof will be sent by: https://github.com/taikoxyz/taiko-mono/blob/main/packages/protocol/contracts/team/proving/ProverSet.sol#L79